templates/data-sources.md.tmpl

```diff
@@ -0,0 +1,35 @@
+---
+# Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Mozilla Public License Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://mozilla.org/MPL/2.0/
+#
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+title: "{{.Name }} {{.Type | lower}}"
+linkTitle: "{{.Name}}"
+page_title: "{{.Name}} {{.Type}} - {{.ProviderName}}"
+subcategory: ""
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+---
+
+# {{.Name }} ({{.Type}})
+
+{{ .Description | trimspace }}
+
+{{ if .HasExample -}}
+## Example Usage
+
+{{ printf "{{tffile %q}}" .ExampleFile }}
+{{- end }}
+
```

Comment:
I see that all examples have 3 files: data-source.tf (resource.tf for sources), terraform.tfvars and variables.tf.
This template will only include data-source.tf (or resources.tf), not the other two files.
In order to include these files, we need to do modifications somewhat along the following lines:
```suggestion

{{ if .HasExample -}}
## Example Usage

variables.tf
{{ tffile ( printf "examples/data-sources/%s/variables.tf" .Name ) }}

terraform.tfvars
{{ tffile ( printf "examples/data-sources/%s/terraform.tfvars" .Name ) }}

main.tf
{{ printf "{{tffile %q}}" .ExampleFile }}
{{- end }}

```
Similarly, the resources.md.tmpl will need to be modified too.